### PR TITLE
[EH] Export tags

### PR DESCRIPTION
**Pull request by aheejin**
This adds exported tags to `exports` section in wasm-emscripten-finalize
metadata so Emscripten can use it.

Also fixes a bug in the parser. We have only recognized the export
format of
```wasm
  (tag $e2 (param f32))
  (export "e2" (tag $e2))
```
and ignored this format:
```wasm
  (tag $e1 (export "e1") (param i32))
```

Companion patch: https://github.com/emscripten-core/emscripten/pull/17064

### CODE REVIEW

File: test/lld/cpp_exception.wat.out [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  "namedGlobals": {
+  },
+  "invokeFuncs": [
+  ],
+  "mainReadsParams": 0,
+  "features": [
+  ]
+}
+-- END METADATA --
+;)
```
Comment (sbc100) [inline test/lld/cpp_exception.wat.out:141]: Could you write a much smaller test in `test/lit/wasm-emscripten-finalize/` the just defines and exports a tag?

I think we are trying to move away from non-lit tests ASAP.

(Sorry I didn't suggest this before you wrote this longer lld test....)

Comment (aheejin) [reply test/lld/cpp_exception.wat.out]: Done. Also fixed a bug in the parser. The previous parser only detected
```wasm
  (tag $e2 (param f32))
  (export "e2" (tag $e2))
```
and couldn't recognize the export in the form of
```wasm
  (tag $e1 (export "e1") (param i32))
```

File: scripts/test/generate_lld_tests.py [hunk 2]
```diff
@@ -89,6 +89,10 @@ def generate_wat_files(llvm_bin, emscripten_root):
         else:
             compile_cmd.append('--target=wasm32-emscripten')
 
+        if 'exception' in src_file:
+            compile_cmd.append('-fwasm-exceptions')
+            link_cmd += ['--export', '__cpp_exception']
+
```
Comment (sbc100) [inline scripts/test/generate_lld_tests.py:95]: I think this can be reverted now

**Review by sbc100 — APPROVED** [review]
lgtm @comment

File: scripts/test/generate_lld_tests.py (hunk 2 above)
Comment (aheejin) [reply scripts/test/generate_lld_tests.py]: Done

**Pull request merged**